civihr_employee_portal/js/reports.js

```diff
@@ -406,6 +406,43 @@
     this.applyFilters();
   };
 
+  /**
+   * Gets the default query parameters for fetching results for the
+   * Leave and absence report. It basically defaults to parameters
+   * for fetching results for the last 30 days starting from current
+   * date.
+   *
+   * @returns {String}
+   */
+  HRReport.prototype.getDefaultFilterQueryForLeaveReport = function () {
+    var toDate = moment().dayOfYear(1).add(1, 'year').subtract(1, 'day').format('YYYY-MM-DD');
+    var fromDate = moment().dayOfYear(1).format('YYYY-MM-DD');
+    var defaultFilterValues = [
+      { name: 'absence_date_filter[min]', value: fromDate },
+      { name: 'absence_date_filter[max]', value: toDate }
+    ];
+
+    return '?' + $.param(defaultFilterValues);
```

Comment:
Any chance this is going to be populated to URL already with `?querystring=smth` ?

Reply:
Nope, there's a check before this function is called, so all good.